### PR TITLE
Add PID class

### PR DESCRIPTION
**Pull request by jballoffet**
# 🎉 New feature

Related: #87

## Summary
This PR improves the PID module by:
- Creating a new class named `PID`.
- Removing the `SetPointInfo` struct in favor of class attributes.
- Turning all global variables into class attributes.
- Removing code duplication for left and right pid controllers.

**Note for the reviewer 1:** #124 needs to be merged and this branch rebased before reviewing this PR.

**Note for the reviewer 2:** the PID module refactoring isn't complete and this is just an stepping stone for other PRs to come. Hacky and ugly code workarounds have been applied on this PR so as to minimize the changes and ease the review process and will be properly solved on a subsequent PR.

More PRs will be sent to keep improving the code structure.

## Test it
Tested by performing manual integration tests on the robot hardware.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: andino_firmware/src/pid.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+namespace andino {
+
+// TODO(jballoffet): Revisit overall logic and add proper documentation.
+class PID {
```
Comment (francocipollone) [inline andino_firmware/src/pid.h:35]: Probably it is the PR to add docs

**Review by francocipollone — COMMENTED** [review]
LGTM overall. I will test it locally(on real robot) before approving

**Comment by jballoffet** [comment]
> LGTM overall. I will test it locally(on real robot) before approving

Great, thanks! Please notice that I have already tested these changes on an actual robot.

**Review by francocipollone — APPROVED** [review]
LGTM!

**Pull request merged**